### ISSUE TITLE
The default value of "minimize" option should be false

### ISSUE THREAD

**Issue by huangxuewu** [Automattic/mongoose#8024]
**Do you want to request a *feature* or report a *bug*?**

The Default value of the minimize option should be false.

**What is the current behavior?**
The default value is true.

**If the current behavior is a bug, please provide the steps to reproduce.**

This default behavior will remove empty Object without any notice.

**What is the expected behavior?**
Empty object should remove only when is intented to do so.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
mongoose 5.6.6 
DB 4.0.10

**Comment by vkarpov15** [comment]
The reason for the minimize option being true is that Mongoose adds `{}` for all nested paths by default:

```javascript
// Without minimize, you would always get `nested: {}`
const schema = Schema({ nested: { name: String } });
```

We'll close this for now because making this change would be very risky and not provide much benefit, but thanks for the suggestion :+1:

**Comment by juona** [comment]
Bumped into this.

Perhaps ideally these should not be related? Or rather, the described behavior seems like a defect. `nested` is an optional field, and so is `name`, so it appears to be valid to provide either `{}` or `{ nested: {} }`. If I don't provide a value for `nested` then nothing should be saved in the DB.

**Comment by vkarpov15** [comment]
I think we should reconsider setting `minimize` to `false` for Mongoose 9, but for different reasons. In `save()`, we need to copy the object using `$toObject()` to apply minimize. If we didn't need to apply minimize, we could just make [this line](https://github.com/Automattic/mongoose/blob/cc60077c043d038fbbb22eb0092e8cb4ffcf761f/lib/model.js#L301) use `_doc`, which would skip a potentially unnecessary cloning of the full document and lead to substantial performance gains on both #14394 and #14897.

@AbdelrahmanHafez @hasezoey do you have any thoughts on this?

**Comment by hasezoey** [comment]
i am partial to this as i have not encountered any problems with the current default (`true`) myself, though i can understand that when coming to mongoose and setting it to a value (even if empty) to not be saved (or expecting the property to exist at all in the saved document (like via mongodb compass)) could be unexpected.

btw is the statement from 2019 still true?:

> The reason for the minimize option being true is that Mongoose adds {} for all nested paths by default:

**Comment by vkarpov15** [comment]
Yes the 2019 statement is still true. If you have a `User` model with `new Schema({ name: { first: String, last: String })`, `const user = new User({})` will still have an empty object for `user.name`. That's because you cannot have getters and setters on `null` or `undefined`, so if we initialized `user.name` to `null` then we would need to do extra work to support change tracking on `user.name = {}; user.name.first = 'John';`.
